testsuite/MDAnalysisTests/coordinates/test_timestep_api.py

```diff
@@ -71,162 +92,34 @@ class BaseTimestepInterfaceTest(TestCase):
-    __test__ = False
-
-    def tearDown(self):
-        del self.ts
-        del self.u
-
-    def test_frame(self):
-        assert_equal(self.ts.frame, 0)
-
-    def test_dt(self):
-        assert_equal(self.u.trajectory.dt, self.ts.dt)
-
-
-class TestCRD(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(XYZ_five, INPCRD)
-        self.ts = u.trajectory.ts
-
-
-class TestDCD(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(PSF, DCD)
-        self.ts = u.trajectory.ts
-
-
-class TestDLPConfig(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(DLP_CONFIG, format='CONFIG')
-        self.ts = u.trajectory.ts
-
-
-class TestDLPHistory(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(DLP_HISTORY, format='HISTORY')
-        self.ts = u.trajectory.ts
-
-
-class TestDMS(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(DMS)
-        self.ts = u.trajectory.ts
-
-
-class TestGMS(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(GMS_ASYMOPT, GMS_ASYMOPT,
-                                  format='GMS', topology_format='GMS')
-        self.ts = u.trajectory.ts
-
-
-class TestGRO(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(GRO)
-        self.ts = u.trajectory.ts
-
-
-class TestINPCRD(BaseTimestepInterfaceTest):
-    __test__ = True
-    def setUp(self):
-        u = self.u = mda.Universe(XYZ_five, INPCRD)
-        self.ts = u.trajectory.ts
-
-
-class TestLAMMPS(BaseTimestepInterfaceTest):
-    __test__ = True
+    @pytest.fixture()
+    def universe(self, topology, trajectory, trajectory_format, topology_format):
+        if trajectory_format is not None and topology_format is not None:
+            return mda.Universe(topology, trajectory, format=trajectory_format,
+                                topology_format=topology_format)
 
-    def setUp(self):
-        u = self.u = mda.Universe(LAMMPSdata)
-        self.ts = u.trajectory.ts
+        if trajectory is not None:
+            return mda.Universe(topology, trajectory)
+        else:
+            return mda.Universe(topology, format=trajectory_format)
 
+    def test_frame(self, universe):
+        assert_equal(universe.trajectory.ts.frame, 0)
 
-class TestLAMMPSDCD(BaseTimestepInterfaceTest):
-    __test__ = True
+    def test_dt(self, universe):
+        assert_equal(universe.trajectory.dt, universe.trajectory.ts.dt)
 
-    def setUp(self):
-        u = self.u = mda.Universe(LAMMPSdata2, LAMMPSdcd2,
-                                  format='LAMMPS', topology_format='DATA',
-                                  timeunit='fs')
-        self.ts = u.trajectory.ts
 
+class TestNCDFBaseTimestepInterface(object):
```

Comment:
Does this separate class solely exist so that you can test for `netCDF4`? If so, this is not necessary anymore. 

Could you then get rid of this class and do everything in the pararmeterized class? (You are already loading ncdf files in the parametrization anyway...).